pyproject.toml

```diff
@@ -46,17 +46,10 @@ pytest-rerunfailures = "^11.1.2"
 coverage = "^7.2.6"
 
 [tool.poetry.group.lint.dependencies]
+ruff = "^0.0.278"
 black = "^23.3.0"
 coverage = "^7.2.6"
-flake8 = "^6.0.0"
-flake8-annotations = "^3.0.1"
-flake8-bugbear = "^23.5.9"
-flake8-tidy-imports = "^4.8.0"
-flake8-pep585 = {version = "^0.1.7", python = ">=3.9"}
-flake8-new-union-types = "^0.4.1"
-flake8-future-annotations = "^1.1.0"
 isort = "^5.12.0"
```

Comment:
Why can't we also replace `isort` with `ruff`?

Reply:
We could, but I think it's a better idea to keep them separate for now, ruff's isort doesn't have complete feature parity with isort in terms of settings, I don't know if we use some of those unsupported settings, but I think we can hold this off for now